README.md

```diff
@@ -180,6 +180,23 @@ Located in the lines above any top-level type that lives within a .cls file, or
     </a>
 </p>
 
+You can create a full revision history by sequencing @author and @since attributes appropriately:
```

Comment:
I would expect:

```
/**
 * This class should be included by `apexdox.engine.includes` setting
 * 
 * @since 2019-01-01 Created
 * @author Peter Weinberg
 * 
 * @since 2022-10-05 Added changelog
 * @author Bill C Riemers
 */
```

To parse exactly the way it does.   The reason being, is Apex, and ApexDox both work in top down order.  If I write:

```
/**
 * Integer foo = bar();
 * @example
 */
```
I don't expect `Integer foo = bar()` to appear in the **example** section.  Because the attribute comes after that text not before.

And in general @ApexDox does not treat blank lines special.  So I would expect it to parse:

```
/**
 * This class should be included by `apexdox.engine.includes` setting
 * 
 * @since 2019-01-01
 * 
 * @author Peter Weinberg
 * @since 2022-10-05 Added changelog
 * 
 * @author Bill C Riemers
 */
```
Exactly the same.  It is obvious to you and me, that in this case the first  `@since` is being used as a version tag, you would have been the one that made the change on 2022-10-05, and I never bothered to indicate a comment or date when I created the class.

But I don't expect the parser to be smart enough to distiquish between these two cases.  In fact I would be very concerned if it did, as it would mean the results would be more prone to unpredictable results.

And actually I do think there is a valid use case where `@author` will come before `@since`.   Any documention created before this update will likely just have one `@author` tag, and one `@since` in random order.   If you look at the test files, I deliberately put the tags out of order in one case, just to make sure the results seem reasonable to me.

Reply:
One thing though that kept coming to mind, is maybe it would be a good idea to have a configuration option to have the old ungrouped parsing of @author and @since.   But I really couldn't come up with a good example of where I thought that was needed, so it seemed like a way to just make it more complicated.

Reply:
Admittedly though, I cannot find anywhere that even documents doxygen works this way.  It just does...

The idea of using `@since` and specifying only a date never made sense to me.  As it is like someone suddenly standing up in the room and shouting a date.   If they don't follow that up about what is significant about that date, I'm likely to dismiss them as a lunatic.   So I was not at all all surprised when I found examples of people using it as a change log in this manner.  But that was at least 12 years ago.  I'd be hard pressed to find those samples again today.

Reply:
I did however, knowingly deviate from even that model abit.  In that if I specify:

```
* @author Bill Riemers
* @author Liam O'Connor
* @since 2021-10-22 Created
* @since 2022-10-21 Added investigations method
* 
* @author Buffy Summers
* @since 2022-01-13 Staked things
```

doxygen would give combine Bill Riemers and Liam O'Connor into a column separated list, under the assumption the both worked the changes that follow together.   

I did not attempt to reproduce that behavour mainly because out of the millions of characters of code Red Hat and our contractors have put in salesforce, not a single developer has found it necessary to list two authors in a row.  I notice even in the cases I know two developer pair coded a solution together, they each add there own `@since` statements.

But if we want to merge the authors together in this case, it wouldn't be to difficult to change the code to do that.  I personally like incremental changes.  So my suggestion is if we want to do that, I do a follow-up request.

Reply:
A slightly better idea.  I make a class for ordered attributes: { name: String, value: String [] }.  So changelog would be an array of ordered attributes.   If the attribute is the same name as the previous, we append to the value array of the last ordered attributes.  If the attribute is different we append a new entry to the ordered attributes array.   Then output could look something like:

**Author(s):** **Peter Weinberg**, **Bill Riemers**
**Since:**
- 2022-10-22 - Pair coded apple attribute
- 2022-10-28 - Added orange attribute
**Author(s):** **Peter Weinberg**
**Since:** 2021-09-15 - Created

This is actually order independant, because we are just outputting in the same order specified.  It is up to the reader to interpret if there is significance to this order.

Notice: In this case I would make `@since` a list only if there is more than one value.  I think this styling could all be done in css, but it has been a few years since I had to do that in css, so if we want that it would probably be a follow-up as it would take several iterations to get it looking really nice.

Reply:
@docbill I'm ok with this suggestion, and I think as long as we're not introducing any behavior that's too unpredictable I'm good with it. I also love the markdown idea, but I do think there's already an extension out there that does it. Let me see if I can find it for you...

Reply:
Also, are you able to run the extension without packaging it first? i.e. in debug mode? I got a new machine since I last worked on this and due to some updates to how vscode extensions are built since then and using the latest version of VS Code I was really struggling to get the extension working in debug mode. In the interest of time I just ended up packaging it and installing from vsix to test out the changes locally.

Reply:
Hate to send you away from my extension, but if you're looking for markdown, perhaps this would work better for you? https://github.com/cesarParra/apexdocs

Reply:
![Screenshot from 2022-10-31-2](https://user-images.githubusercontent.com/3169368/199051320-b6f3e38d-57c1-4041-9e6b-6d69f44751d8.png)
![Screenshot from 2022-10-31-1](https://user-images.githubusercontent.com/3169368/199051349-c1a5ee0d-f6ee-4fe2-b6b8-fd7a10648fe8.png)

Reply:
It is not quite as pretty as a change log.  But the code is simple, and it takes care of the order dependence by simply preserving order, and not assuming the order has any particular meaning.   Other than of course we are grouping like tags together only until there is a non-like tag.

Reply:
In regards to running in debugging.  I managed to succeed in that once, and having managed to do it again using what I thought was the exact same sequence.  However, the one time was enough for me to figure out the usage of the snapshots.